File: remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
  let i = 0;
  while (i < nums.length) {
    if (nums[i] === val && i !== 0) {
      nums = nums.slice(0, i).concat(nums.slice(i + 1));
    } else if (nums[i] === val && i == 0) {
      nums = nums.slice(1);
    }
    i++;
  }
  while (i > 0) {
    if (nums[i] === val && i !== nums.length - 1) {
      nums = nums.slice(0, i).concat(nums.slice(i + 1));
    } else if (nums[i] === val && i == nums.length) {
      nums = nums.slice(0, i - 1);
    }
    i--;
  }

  return nums;
};

File: remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int front = 0;
        int back = nums.length;
        while (front < back) {
            if (nums[front] == val) {
                nums[front] = nums[back - 1];
                back--;
            } else {
                front++;
            }
        }
        return back;
    }
}